esmvaltool/diag_scripts/droughts/diffmap.py

```diff
@@ -0,0 +1,356 @@
+    ):
+        cube.data[:, 359] = cube.data[:, 0]
+
+
+def plot(
+    cfg: dict,
+    meta: dict,
+    cube: Cube,
+    basename: str,
+    kwargs: dict | None = None,
+) -> None:
+    """Plot map using diag_scripts.shared module."""
+    plotfile = e.get_plot_filename(basename, cfg)
+    plot_kwargs = cfg.get("plot_kwargs", {}).copy()
+    if kwargs is not None:
+        plot_kwargs.update(kwargs)
+    if "vmax" in plot_kwargs and "vmin" in plot_kwargs:
+        plot_kwargs["levels"] = np.linspace(
+            plot_kwargs["vmin"],
+            plot_kwargs["vmax"],
+            9,
+        )
+    label = plot_kwargs.get("cbar_label", "{short_name} ({units})")
+    plot_kwargs["cbar_label"] = label.format(**meta)
+    for coord in cube.coords(dim_coords=True):
+        if not coord.has_bounds():
+            log.warning("NO BOUNDS. GUESSING: %s", coord.name())
+            cube.coord(coord.name()).guess_bounds()
+    fill_era5_gap(meta, cube)
+    add_cyclic_point(cube.data, cube.coord("longitude").points)
+    mapplot = e.plot.global_contourf(cube, **plot_kwargs)
+    if cfg.get("clip_land", False):
+        plt.gca().set_extent((220, 170, -55, 90))  # type: ignore[attr-defined]
+    plt.title(meta.get("title", basename))
+    if cfg.get("strip_plots", False):
+        plt.gca().set_title(None)
+        plt.gca().set_ylabel(None)
+        plt.gca().set_xlabel(None)
+        cb_mappable = mapplot.colorbar.mappable
+        mapplot.colorbar.remove()
+        plot_colorbar(cfg, plotfile, plot_kwargs, mappable=cb_mappable)
+    fig = mapplot.get_figure()
+    fig.savefig(plotfile, bbox_inches="tight")
+    plt.close()
+    log.info("saved figure: %s", plotfile)
+
+
+def apply_plot_kwargs_overwrite(
+    kwargs: dict,
+    overwrites: dict,
+    metric: str,
+    group: str,
+) -> dict:
+    """Apply plot_kwargs_overwrite to kwargs dict for selected plots."""
+    for overwrite in overwrites:
+        new_kwargs = overwrite.copy()
+        groups = new_kwargs.pop("group", [])
+        if not isinstance(groups, list):
+            groups = [groups]
+        if len(groups) > 0 and group not in groups:
+            continue
+        metrics = new_kwargs.pop("metric", [])
+        if not isinstance(metrics, list):
+            metrics = [metrics]
+        if len(metric) > 0 and metric not in metrics:
+            continue
+        kwargs.update(new_kwargs)
+    return kwargs
+
+
+def calculate_diff(cfg, meta, mm_data, output_meta, group) -> None:
+    """Absolute difference between first and last years of a cube.
+
+    Calculates the absolut and relative difference between the first and last
+    period of a cube. Write data to mm and optionally plot each dataset.
+    """
+    cube = iris.load_cube(meta["filename"])
+    if meta["short_name"] in cfg.get("convert_units", {}):
+        pp.convert_units(cube, cfg["convert_units"][meta["short_name"]])
+    with contextlib.suppress(Exception):
+        # TODO: maybe fix this within cmorizer
```

Comment:
```suggestion
        # TODO: maybe fix this within cmorizer
```
Maybe make an issue for this so it doesn't get lost?